uvicorn/lifespan/on.py

```diff
@@ -84,6 +84,11 @@ async def main(self) -> None:
                 "state": self.state,
             }
             await app(scope, self.receive, self.send)
+        except asyncio.CancelledError:
+            # Lifespan task was cancelled, likely due to the server shutting down.
+            # This is not an error that should be handled by BaseException above,
+            # so we just log it and exit.
+            self.logger.info("Lifespan task cancelled.")
```

Comment:
[`CancelledError` should generally be re-raised](https://docs.python.org/3/library/asyncio-exceptions.html#asyncio.CancelledError), in order to propagate cancellation up the call chain or to the to-be-cancelled task.
```suggestion
            self.logger.info("Lifespan task cancelled.")
            raise
```

Reply:
You are correct. Initially I went with the reraise, but the issue persisted, so I naively assumed it was already the top of the chain and that not propagating it would be fine.

I think the main reason is because the issue is not actually here but on starllete (I commented about in https://github.com/encode/uvicorn/pull/2367#issuecomment-2168196065)